debugger/symbols.ml

```diff
@@ -37,6 +37,18 @@ let events_by_module =
 let all_events_by_module =
   (Hashtbl.create 17 : (string, debug_event list) Hashtbl.t)
 
+let partition_modules evl =
+  let rec partition_modules' ev evl =
+    match evl with
+      [] -> [ev],[]
+    | ev'::evl ->
+       let evl,evll = partition_modules' ev' evl in
+       if ev.ev_module = ev'.ev_module then ev::evl,evll else [ev],evl::evll
+  in
+  match evl with
+    [] -> []
+  | ev::evl -> let evl,evll = partition_modules' ev evl in evl::evll
```

Comment:
Exactly, it assumes that they are adjacent. But maybe should I have been clearer. This PR is not a "request". This is just what I did in my local version to resolve our problem. Please of course do as you think it is good to do. Our request for the software that we are developing is before all that ocamldebug works with -pack (and actually, even better, that it also works in dynamically linked modules). With all our support!